sdk/python/test/e2e/conftest.py

```diff
@@ -0,0 +1,21 @@
+import os
+import pytest
+
+from kubeflow.training.utils.utils import is_running_in_k8s
+from kubeflow.training import TrainingClient
+
+
+def pytest_addoption(parser):
+    parser.addoption("--namespace", action="store", default="default")
+
+
+@pytest.fixture
+def job_namespace(request):
+    return request.config.getoption("--namespace")
+
+@pytest.fixture
+def training_client():
+    if is_running_in_k8s():
+        return TrainingClient()
+    else:
+        return TrainingClient(config_file=os.getenv("KUBECONFIG", "~/.kube/config"))
```

Comment:
Why do we need this client?  We can use the TrainingClient since the existing client automatically switches to in cluster client: 

https://github.com/kubeflow/training-operator/blob/1cdf4eefe90082b2371ed717103fcb8509896acd/sdk/python/kubeflow/training/api/training_client.py#L54-L60

Reply:
Since we're passing a default config file (`"~/.kube/config"`) in the existing client code, it never switches to in-cluster config. 
`if config_file` condition will evaluate to True and the `load_kube_config` function will be called. 
https://github.com/kubeflow/training-operator/blob/master/sdk/python/test/e2e/test_e2e_tfjob.py#L39

LMK if I'm missing something.

Reply:
IIUC, If we don't pass a config file to the TrainingClinet and we run tests out of the cluster, config.load_kube_config uses KUBECONFIG or `~/.kube/config`. So I think we can use the TrainingClient instead of this function. 

https://github.com/kubernetes-client/python-base/blob/060cac10e53169c904e0d50b7448233829019e35/config/kube_config.py#L782-L798

https://github.com/kubernetes-client/python-base/blob/060cac10e53169c904e0d50b7448233829019e35/config/kube_config.py#L48

Reply:
makes sense. We don't need `training_client` fixture. Will remove it.